### PR TITLE
feature: add IP over Infiniband interface support

### LINKED ISSUE #52: Add support to IP over Infiniband interface

### PR DESCRIPTION
**Pull request by ffmancera**
Signed-off-by: Fernando Fernandez Mancera <ffmancera@riseup.net>

### CODE REVIEW

**Comment by ffmancera** [comment]
This needs a release from rtnetlink.

**Comment by ffmancera** [comment]
Waiting for https://github.com/little-dude/netlink/issues/119

File: src/lib/netlink/ipoib.rs [hunk 1]
```diff
@@ -0,0 +1,4 @@
+use crate::IpoibInfo;
+use netlink_packet_route::rtnl::link::nlas::InfoIpoib;
+
+pub(crate) fn parse_ipoib_info(infos: &[Info]
```
Comment (cathay4t) [inline src/lib/netlink/ipoib.rs]: This file is not finished?

**Review by cathay4t — CHANGES_REQUESTED** [review]

File: src/lib/netlink/ipoib.rs (hunk 1 above)
Comment (ffmancera) [reply src/lib/netlink/ipoib.rs]: I've removed. That was initial work not needed any more. `parse_ipoib_info` is now done in `src/lib/ifaces/ipoib.rs` as we are using `InfoIpoib` from `netlink_packet_route` and netlink parsing is not needed.

Everything is updated now. Sorry for this.

**Comment by ffmancera** [comment]
As this depends on #95 is stuck again

**Review by cathay4t — CHANGES_REQUESTED** [review]
With kernel-4.18.0-282.el8.x86_64, I got this warnning:

```
Unknown IPoIB info Other(DefaultNla { kind: 3, value: [0, 0] })
```

Please investigate.

**Pull request closed without merging**